### PR TITLE
fix: memory leak when `writeToDisk` used

### LINKED ISSUE #469: `writeToDisk` leads the compile time keep increasing.
* Operating System: MacOS 10.15
* Node Version: v10.15.2
* NPM Version: 6.11.3
* webpack Version: 4.40.2
* webpack-dev-middleware Version: 3.8.1

### Expected Behavior
The compile-time should not keep increasing.

### Actual Behavior

The compile-time keep increasing, and log shows duplicate write to disk keep increasing.
e.g. First compile:
```
[16:31:46] ➤ ｢wdm｣: Asset written to disk: app/js/app.js
writeToDisk
[16:31:46] ➤ ｢wdm｣: Asset written to disk: app/js/app.js.map
writeToDisk
[16:31:46] ➤ ｢wdm｣: Asset written to disk: app/js/app.js
[16:31:46] ➤ ｢wdm｣: Asset written to disk: app/js/app.js.map
```
Third compile:
```
[16:33:01] ➤ ｢wdm｣: Asset written to disk: app/js/app.js
writeToDisk
[16:33:01] ➤ ｢wdm｣: Asset written to disk: app/js/app.js.map
writeToDisk
[16:33:01] ➤ ｢wdm｣: Asset written to disk: app/js/app.js
writeToDisk
[16:33:01] ➤ ｢wdm｣: Asset written to disk: app/js/app.js.map
writeToDisk
[16:33:01] ➤ ｢wdm｣: Asset written to disk: app/js/app.js
writeToDisk
[16:33:01] ➤ ｢wdm｣: Asset written to disk: app/js/app.js
[16:33:01] ➤ ｢wdm｣: Asset written to disk: app/js/app.js.map
writeToDisk
[16:33:01] ➤ ｢wdm｣: Asset written to disk: app/js/app.js.map
```

### Code

```js
          webpackDevMiddleware(compiler, {
            publicPath: webpackConfig.dev.output.publicPath,
            stats: { colors: true },
            writeToDisk: true,
            logTime: true,
            logLevel: 'debug'
          }),
```

### How Do We Reproduce?
I think to change the `logLevel` to `debug` will see the log for any project.

### PR DESCRIPTION
**Pull request by evilebottnawi**
This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

fixes #469

### Breaking Changes

No

### Additional Info

Need refactor test, let's do it in future
Need fix compatibility with webpack@5, i will do it in next PR

### CODE REVIEW

File: lib/fs.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ module.exports = {
 
     for (const compiler of compilers) {
       compiler.hooks.emit.tap('WebpackDevMiddleware', (compilation) => {
+        if (compiler.hasWebpackDevMiddlewareAssetEmittedCallback) {
```
Comment (evilebottnawi) [inline lib/fs.js]: Ugly name, but we guarantee what we don't intersect with something

Comment (evilebottnawi) [reply lib/fs.js]: Also in webpack@5 `assetEmitted` hook has `compilation` in arguments so we simplify this code in near future

**Comment by evilebottnawi** [comment]
@hiroppy feel free to feedback

File: lib/fs.js (hunk 1 above)
Comment (unbug) [reply lib/fs.js]: `compiler.hasWebpackDevMiddlewareAssetEmitted` is enough to make the point.

File: lib/fs.js [hunk 2]
```diff
@@ -74,6 +78,7 @@ module.exports = {
             });
           }
         );
+        compiler.hasWebpackDevMiddlewareAssetEmittedCallback = true;
```
Comment (unbug) [inline lib/fs.js]: We don't actually need to create a flag for the `compiler`. e.g.
```
let hasWebpackDevMiddlewareAssetEmitted = false;
compiler.hooks.emit.tap('WebpackDevMiddleware', (compilation) => {
// ...

hasWebpackDevMiddlewareAssetEmitted = true;

```

Comment (evilebottnawi) [reply lib/fs.js]: @unbug developers can add new compiler in runtime

**Pull request merged**